File: app/src/main/java/com/example/android/newsapp/refactor/mvp/base/BasePresenter.java
package com.example.android.newsapp.refactor.mvp.base;

public abstract class BasePresenter<V extends BaseView> {

    protected V view;

    public final void onTakeView(V view) {
        this.view = view;
        onTakenView();
    }

    public final void onDropView() {
        this.view = null;
    }

    public void onTakenView() {
        //This method will be implement by children
    }
}

File: app/src/main/kotlin/com/example/android/newsapp/mvp/base/BasePresenter.kt
package com.example.android.newsapp.mvp.base


abstract class BasePresenter<V : BaseView> {

    protected var view: V? = null

    fun onTakeView(view: V) {
        this.view = view
        onTakenView()
    }

    fun onDropView() {
        this.view = null
    }

    open fun onTakenView() {
        //This method will be implement by children
    }
}
